File: 0344-reverse-string/0344-reverse-string.cpp
class Solution {
public:
    
    // TLE
    
    /*void reverseString(vector<char>& s) {
        char temp;
        for(int i =0; i<s.size(); i++){
            for(int j=i+1; j<s.size(); j++){
                temp = s[i];
                s[i] = s[j];
                s[j] = temp;
            }
        }
    }*/

    
    
    vector<char> reverseString(vector<char>& s){
        vector<char> res;
        for(int i=s.size()-1; i!=-1; i--){
            res.push_back(s[i]);
        }
        for(int i=0; i < res.size(); i++){
            s[i] = res[i];
        } 
        return s;
    }
    
    
    /*void reverseString(vector<char>& s){
        char temp;
        for(int i=0; i<s.size(); i++){
            for(int j=i+1; j<s.size(); j++){
                //temp = s[i];
                //s[i] = s[j];
                //s[j] = temp;
                swap(s[i],s[j]);
            }
        }
    }*/
};

File: 0344-reverse-string/0344-reverse-string.java
class Solution {

    // O(n^2)
    /*public void reverseString(char[] s) {
        char temp;
        for(int i=0; i<s.length; i++){
            for(int j=i+1; j<s.length; j++){
                temp = s[i];
                s[i] = s[j];
                s[j] = temp;
            }
        }
    }*/

    // O(n)
    public void reverseString(char[] s) {
        char ar[];
        ar = new char[s.length];

        for(int i = s.length-1; i!=-1; i--){
            ar[Math.abs(i-s.length+1)] = s[i];
        }
        for(int i=0; i<s.length; i++){
            s[i] = ar[i];
        }


    }
}